components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -15,25 +17,36 @@ import SearchIcon from 'components/widgets/icons/search_icon';
 
 import ChannelRow from './channel_row';
 
-export default class ChannelList extends React.PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            searchAllChannels: PropTypes.func.isRequired,
-            getData: PropTypes.func.isRequired,
-        }).isRequired,
-        data: PropTypes.array,
-        total: PropTypes.number,
+interface ChannelListProps {
+    actions: {
+        searchAllChannels: (term: string, notAssociatedToGroup?: string, excludeDefaultChannels?: boolean, page?: number, perPage?: number) => ActionFunc | ActionResult;
+        getData: (page: number, perPage: number, notAssociatedToGroup? : string, excludeDefaultChannels?: boolean) => ActionFunc | ActionResult;
+        removeGroup?: () => void;
```

Comment:
I don't think you need this Prop, it's not passed in through Redux

Reply:
> Thanks @allenlai18, this is looking great!
> As part of this ticket, can you also convert the redux connector to TypeScript as well?

you mean index.js right?  yes i will do that